src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/Publishers.cs

```diff
@@ -11,87 +11,39 @@ public class Publishers
     {
         internal IEnumerable<PublisherAddress> GetPublisherFor(Type eventType)
         {
-            List<PublisherAddress> staticPublishersForType;
-            if (!staticPublishers.TryGetValue(eventType, out staticPublishersForType))
-            {
-                staticPublishersForType = emptyList;
-            }
-
-            if (dynamicRules.Count > 0)
-            {
-                var dynamicAddresses = new List<PublisherAddress>();
-                foreach (var rule in dynamicRules)
-                {
-                    var address = rule(eventType);
-                    if (address != null)
-                    {
-                        dynamicAddresses.Add(address);
-                    }
-                }
-
-                return staticPublishersForType.Count > 0 ? staticPublishersForType.Concat(dynamicAddresses) : dynamicAddresses;
-            }
-
-            return staticPublishersForType;
-        }
-
-        /// <summary>
-        /// Registers a logical publisher for a given event type.
-        /// </summary>
-        /// <param name="eventType">The event type.</param>
-        /// <param name="publisher">The publisher endpoint.</param>
-        public void Add(Type eventType, string publisher)
-        {
-            AddStaticPublisher(eventType, PublisherAddress.CreateFromEndpointName(publisher));
+            HashSet<PublisherAddress> addresses;
+            return publishers.TryGetValue(eventType, out addresses)
+                ? addresses
+                : Enumerable.Empty<PublisherAddress>();
         }
 
         /// <summary>
-        /// Registers a publisher endpoint instance for a given event type.
+        /// Adds or replaces a set of publisher registrations.
```

Comment:
I think we should improve the docs a little bit to mention that this will be scoped to the source key which has to be reused if you want to replace entries?